lifecycle-coroutines/src/commonTest/kotlin/com/arkivanov/essenty/lifecycle/coroutines/LifecycleCoroutinesExtTest.kt

```diff
@@ -0,0 +1,202 @@
+
+    @AfterTest
+    fun afterTesting() {
+        Dispatchers.resetMain()
+    }
+
+    @Test
+    fun test_passed_state_CREATED_must_be_trigger_block_once() = runTest {
+        val testState = Lifecycle.State.CREATED
+        val expected = listOf(testState.name)
+
+        val actual = executeRepeatOnEssentyLifecycleTest(testState)
+        advanceUntilIdle()
+
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun test_passed_state_STARTED_must_be_trigger_block_twice() = runTest {
+        val testState = Lifecycle.State.STARTED
+        val expected = listOf(testState.name, testState.name)
+
+        val actual = executeRepeatOnEssentyLifecycleTest(testState)
+        advanceUntilIdle()
+
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun test_passed_state_RESUMED_must_be_trigger_block_twice() = runTest {
+        val testState = Lifecycle.State.RESUMED
+        val expected = listOf(testState.name, testState.name)
+
+        val actual = executeRepeatOnEssentyLifecycleTest(testState)
+        advanceUntilIdle()
+
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun test_passed_state_DESTROYED_must_not_be_trigger_block() = runTest {
+        val testState = Lifecycle.State.DESTROYED
+        val expected = emptyList<String>()
+
+        val actual = executeRepeatOnEssentyLifecycleTest(testState)
+        advanceUntilIdle()
+
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun test_flow_passed_state_CREATED_must_be_trigger_block_once() = runTest {
+        val state = Lifecycle.State.CREATED
+        val expect = listOf(state.name, state.name)
+
+        val actual = executeFlowWithEssentyLifecycleTest(state)
+        advanceUntilIdle()
+
+        assertEquals(expect, actual)
+    }
+
+    @Test
+    fun test_flow_passed_state_STARTED_must_be_trigger_block_twice() = runTest {
+        val state = Lifecycle.State.STARTED
+        val expect = listOf(state.name, state.name, state.name, state.name)
+
+        val actual = executeFlowWithEssentyLifecycleTest(state)
+        advanceUntilIdle()
+
+        assertEquals(expect, actual)
+    }
+
+    @Test
+    fun test_flow_passed_state_RESUMED_must_be_trigger_block_twice() = runTest {
+        val state = Lifecycle.State.RESUMED
+        val expect = listOf(state.name, state.name, state.name, state.name)
+
+        val actual = executeFlowWithEssentyLifecycleTest(state)
+        advanceUntilIdle()
+
+        assertEquals(expect, actual)
+    }
+
+    @Test
+    fun test_flow_passed_state_DESTROYED_must_not_be_trigger_block() = runTest {
+        val state = Lifecycle.State.DESTROYED
+        val expect = emptyList<String>()
+
+        val actual = executeFlowWithEssentyLifecycleTest(state)
+        advanceUntilIdle()
+
+        assertEquals(expect, actual)
+    }
+
+    private suspend fun executeRepeatOnEssentyLifecycleTest(
+        lifecycleState: Lifecycle.State,
+        onEventAppears: () -> String = { lifecycleState.name }
```

Comment:
Looks like the argument is never passed. Remove?

Reply:
Deleted.